sleap/gui/dialogs/filedialog.py

```diff
@@ -62,7 +62,7 @@ def save(cls, *args, **kwargs):
         # Make sure filename has appropriate file extension.
         if is_non_native and filter:
             # Get ext from selected filter (i.e., file type).
-            match = re.search("\\(\\*(\\.[a-z]+)", filter)
+            match = re.search("\\(\\*(\\.[a-zA-Z0-9]+)\\)", filter)
             if match:
                 filter_ext = match[1]
```

Comment:
Thanks!
I made the change (with a small change in the regex pattern to catch only the extension without the star).

But for the last line, I left `filename = f"{filename}{filter_ext}"`, that seems better to me to add the correct extension than changing the file extension. Especially given that Sleap outputs filenames with dots in it, part of the filename could be deleted because it was considered the extension.